src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/publicip/v2/OpenStackPublicIpPluginTest.java

```diff
@@ -1,576 +1,1231 @@
+        Mockito.doThrow(expectedException).when(this.client).doPostRequest(Mockito.eq(endpoint),
+                Mockito.eq(request.toJson()), Mockito.eq(cloudUser));
+
+        PowerMockito.mockStatic(OpenStackHttpToFogbowExceptionMapper.class);
+        PowerMockito.doCallRealMethod().when(OpenStackHttpToFogbowExceptionMapper.class, TestUtils.MAP_METHOD,
+                Mockito.any());
+
+        try {
+            // exercise
+            this.plugin.doCreateSecurityGroup(instanceId, cloudUser);
+            Assert.fail();
+        } catch (Exception e) {
+            // verify
+            PowerMockito.verifyStatic(OpenStackHttpToFogbowExceptionMapper.class, Mockito.times(TestUtils.RUN_ONCE));
+            OpenStackHttpToFogbowExceptionMapper.map(Mockito.eq(expectedException));
+        }
+    }
+    
+    // test case: When calling the doCreateSecurityGroupResponseFrom method with a
+    // JSON malformed, it must verify that a UnexpectedException was throw.
+    @Test
+    public void testDoCreateSecurityGroupResponseFromJsonMalformed() throws FogbowException {
+        // set up
+        String json = TestUtils.JSON_MALFORMED;
+        String expected = String.format(Messages.Error.ERROR_WHILE_CREATING_RESOURCE_S,
+                OpenStackCloudUtils.SECURITY_GROUP_RESOURCE);
+        try {
+            // exercise
+            this.plugin.doCreateSecurityGroupResponseFrom(json);
+            Assert.fail();
+        } catch (UnexpectedException e) {
+            // verify
+            Assert.assertEquals(expected, e.getMessage());
+        }
+    }
+    
+    // test case: When calling the doRequestInstance method, it must verify
+    // that the call was successful.
+    @Test
+    public void testDoRequestInstance() throws Exception {
+        // set up
+        OpenStackV3User cloudUser = this.testUtils.createOpenStackUser();
+        
+        String endpoint = NEUTRON_PREFIX_ENDPOINT
+                + OpenStackCloudUtils.NETWORK_V2_API_ENDPOINT 
+                + OpenStackPublicIpPlugin.FLOATINGIPS;
+        
+        CreateFloatingIpRequest request = new CreateFloatingIpRequest.Builder()
+                .floatingNetworkId(FAKE_FLOATING_NETWORK_ID)
+                .portId(FAKE_NETWORK_PORT_ID)
+                .projectId(TestUtils.FAKE_PROJECT_ID)
+                .build();
+        
+        String json = FAKE_CREATE_FLOATING_IP_FROM_JSON_RESPONSE;
+        Mockito.when(this.client.doPostRequest(Mockito.eq(endpoint), Mockito.eq(request.toJson()), Mockito.eq(cloudUser)))
+                .thenReturn(json);
 
-		PowerMockito.mockStatic(SharedOrderHolders.class);
-		BDDMockito.given(SharedOrderHolders.getInstance()).willReturn(this.sharedOrderHolders);
+        CreateFloatingIpResponse response = CreateFloatingIpResponse.fromJson(json);
+        PowerMockito.mockStatic(CreateFloatingIpResponse.class);
+        PowerMockito.when(CreateFloatingIpResponse.fromJson(Mockito.eq(json))).thenReturn(response);
```

Comment:
Are these 3 lines equivalent to mockStatic + callRealMethod?

Reply:
Okay, I'll fix it.